### ISSUE TITLE
Powershell implicit conversion happening when returning an array with a single object (should be fixed in 3.0) and could not use .count 

### ISSUE THREAD

**Issue by pingramk** [PowerShell/PowerShell#4007]
Steps to reproduce
------------------
```powershell
function test {

    $tempary = @()
    
    $tempitem = New-Object PSCustomObject

    $tempItem | Add-Member -MemberType NoteProperty -Name "asdf" -Value "testing in prod"
    
    $tempary += $tempItem

    return $tempary
}

&{

    $t = test

    if( $t.Count -gt 0){
        Write-Host "Ye got the count"
    } else {
        Write-Host "No Count"
    }
} 
```

Expected behavior
-----------------

Array object. So .count could be used to check.

Actual behavior
---------------

Implicit conversion from array to single object though the function returns an array

Environment data
----------------

```powershell
PS C:\Users\rkomand> $PSVersionTable

Name                           Value
----                           -----
PSVersion                      5.1.15063.296
PSEdition                      Desktop
PSCompatibleVersions           {1.0, 2.0, 3.0, 4.0...}
BuildVersion                   10.0.15063.296
CLRVersion                     4.0.30319.42000
WSManStackVersion              3.0
PSRemotingProtocolVersion      2.3
SerializationVersion           1.1.0.1
```

**Comment by SteveL-MSFT** [comment]
Repros with PSCore6 beta2

**Comment by mklement0** [comment]
This behavior may be surprising, but I think it is by design:

Collections sent to the success output stream are _unwrapped_ and sent item by item.

When you capture the result of a pipeline in a variable, a single item is stored  as-is, whereas multiple items are stored as a `[System.Object[]]` arrays - irrespective of the type of the collection sent to the output stream.

```powershell
# Even though an *array list* is output, the result is a *regular array*
> (& { [system.collections.arraylist] (1, 2) }).GetType().Name 
Object[]
```

To send a collection _as a whole_ to the output stream, wrap it in an array:

```powershell
# Wrap the array list in a single-item array with ",", the array-construction operator,
# to output the collection as a whole.
> (& { , [system.collections.arraylist] (1, 2) }).GetType().Name 
ArrayList
```

On a side note: the code sample relies on a _bug_ that will be fixed - see #3671 
Since v3, _all_ scalars should have a `.Count` property (added by PS on demand), to unify the handling of scalars and arrays.

```powershell
> ([pscustomobject]::new()).Count
 # !! bug - no output

> (666).Count
1  # OK - a scalar has a count of 1
```

**Comment by SteveL-MSFT** [comment]
Thanks @mklement0 the unwrapping is definitely `by design` and will continue to bring future discussion.  The lack of `.count` is and issue and dupe of yours.
